[Docs] Add new_task example; improve: feature store engines descriptions, log_artifact, function handler (#4756)
mlrun/mlrun: docs/runtimes/images.md

The images are:

> - `mlrun/mlrun`: An MLRun image includes preinstalled OpenMPI. Useful as a base image for simple jobs.

Reviewer: ```suggestion
- `mlrun/mlrun`: An MLRun image includes preinstalled OpenMPI and other ML packages. Useful as a base image for simple jobs.
```

Author: updated

- `mlrun/mlrun-gpu`: The same as `mlrun/mlrun` but for GPUs, including Open MPI. 

> - `mlrun/ml-base`: The image for file acquisition, compression, dask jobs, simple training jobs and other utilities. Like `mlrun/mlrun` with the addition of Miniconda and other python packages.

Reviewer: Not sure how file acquisition or compression are specifically related to this image. Also, mlrun/mlrun is not based on miniconda as well.
```suggestion
- `mlrun/ml-base`: Similar to `mlrun/mlrun` with the addition of Dask and other python packages.
```

Author: updated

- `mlrun/jupyter`: An image with Jupyter giving a playground to use MLRun in the open source. Built on top of jupyter/scipy-notebook, with the addition of MLRun and several demos and examples.
- `mlrun/mlrun-api`: The image used for running the MLRun API.

> - `mlrun/mlrun-ui`: The image used for running the MLRun UI.

Reviewer: These 2 are not related to runtimes images. Don't think they should be here.

Author: deleted, and changed the heading name to **MLRun runtime images**
